pillar/vault.py

```diff
@@ -169,6 +169,33 @@ def _authenticate(conn):
         conn.token = os.environ.get('VAULT_TOKEN')
 
 
+def couple(location, conn):
+    """
+    If location is a dictionary, return a dictionary of k->v pairs looked up from vault.
+    If location is a string, return the value looked up from vault.
+    """
+    coupled_data = {}
```

Comment:
I'm confused as to why this the function and this are named 'couple', maybe using the word in the docstring in the description would help.... or change it to something like walk_secrets_namespace or something descriptive